File: php/test/Mockist/MyClassTest.php
<?php

namespace Foo\Test\Mockist;

use Foo\Mockist\Collaborator;
use Foo\Mockist\MyClass;
use Prophecy\PhpUnit\ProphecyTestCase;

class MyClassTest extends ProphecyTestCase
{
    /**
     * Mock
     * @test
     */
    public function shouldUseTheExternalCollaborator()
    {
        $myCollaboratorProphecy = $this->prophesize('Foo\Mockist\Collaborator');
        /** @var Collaborator $collaborator */
        $collaborator = $myCollaboratorProphecy->reveal();
        $myClass = new MyClass($collaborator);

        $myClass->run();

        $myCollaboratorProphecy->collaborate()->shouldBeCalled();
    }

    /**
     * Stub
     * @test
     */
    public function shouldReturnTheCollaboratorResponse()
    {
        $myCollaboratorProphecy = $this->prophesize('Foo\Mockist\Collaborator');
        $collaboratorResponse = 'collaborator response';
        $myCollaboratorProphecy->collaborate()->willReturn($collaboratorResponse);
        /** @var Collaborator $collaborator */
        $collaborator = $myCollaboratorProphecy->reveal();
        $myClass = new MyClass($collaborator);

        $response = $myClass->run();

        $this->assertEquals($collaboratorResponse, $response);
    }
}

File: java/test/Mockist/MyClassTest.java
package Mockist;

import static org.mockito.Mockito.*;
import static org.junit.Assert.*;

import org.junit.Test;

public class MyClassTest {

	@Test
	public void shouldInteractWithTheMock() {
		Collaborator collaborator = mock(Mockist.Collaborator.class);		
		MyClass myClass = new MyClass(collaborator);
		
		myClass.run();
		
		verify(collaborator).collaborate();
	}
	
	@Test
	public void shouldRetrieveTheStubResponse(){
		Collaborator collaborator = mock(Mockist.Collaborator.class);		
		String collaboratorResponse = "collaborator response";
		when(collaborator.collaborate()).thenReturn(collaboratorResponse);
		MyClass myClass = new MyClass(collaborator);
		
		String result = myClass.run();
		
		assertEquals(result, collaboratorResponse);
		}
}

